GameConfiguration.h

```diff
@@ -13,7 +13,7 @@ struct GameConfiguration {
     Config config;
 
     // items not configurable via the config file
-    bool overlay_mode;
+    bool immersive_mode;
```

Comment:
could you move this (and related code) to the immersive mode CL? I'd like to see this PR just remove code